src/Microsoft.Data.Sqlite/SqliteCommand.cs

```diff
@@ -111,11 +114,16 @@ public new virtual SqliteDataReader ExecuteReader(CommandBehavior behavior)
             do
             {
                 Sqlite3StmtHandle stmt;
-                var rc = NativeMethods.sqlite3_prepare_v2(
-                    Connection.DbHandle,
-                    tail,
-                    out stmt,
-                    out tail);
+                var rc = NativeMethods.sqlite3_prepare_v2_blocking(
+                        Connection.DbHandle,
+                        tail,
+                        out stmt,
+                        out tail,
+                        CommandTimeout*1000);
+                if (rc == SQLITE_BUSY)
+                {
+                    throw new TimeoutException(Strings.FormatCommandTimeout(CommandTimeout));
```

Comment:
Don't throw a `TimeoutException`; anything other than a `SqliteException` would be unexpected to callers. Just let the next line throw.